### PR TITLE
3173 move mobile entity navigation

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Split out EntityNav into a new component
- Refactor Region Screen to be functional component
- Add back button
---

### Screenshots:
See issue

### CODE REVIEW

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+          currentFilter={selectedFilter}
+          onFilterChange={onChangeMeasure}
+          filterIsExpanded={measureFilterIsExpanded}
+          onFilterOpen={onToggleMeasureExpand}
+          onFilterClose={onToggleMeasureExpand}
+          showLoadingIcon={isMeasureLoading}
+          theme={{ background: WHITE, color: '#000' }}
+        />
+        {mobileListItems.length === 0 && !isLoading && (
+          <NoResultsText>No clinics found in this region.</NoResultsText>
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:90]:
```suggestion
          <NoResultsText>No children found within the current entity.</NoResultsText>
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  const mobileListItems = getListItemsFromOrganisationUnitChildren(
+    selectOrgUnitChildren(state, orgUnit.organisationUnitCode),
+    isMeasureLoading,
+    measureInfo,
+  );
+
+  const title = mobileListItems.some(i => i && i.type === ENTITY_TYPE.FACILITY)
+    ? 'Browse Facilities'
+    : 'Browse Districts';
+
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:209]:
```suggestion
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  return {
+    orgUnit,
+    mobileListItems,
+    measureFilters,
+    selectedFilter,
+    measureFilterIsExpanded: isExpanded,
+    isLoading: isLoadingOrganisationUnit,
+    isMeasureLoading,
+    title,
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:224]:
```suggestion
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+  isMeasureLoading,
+  measureFilters,
+  measureFilterIsExpanded,
+  mobileListItems,
+  onChangeMeasure,
+  onChangeOrgUnit,
+  onToggleMeasureExpand,
+  orgUnit,
+  selectedFilter,
+  title,
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:58]:
```suggestion
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+  isMeasureLoading: PropTypes.bool,
+  measureFilters: PropTypes.array,
+  measureFilterIsExpanded: PropTypes.bool,
+  mobileListItems: PropTypes.array,
+  orgUnit: PropTypes.object.isRequired,
+  onToggleMeasureExpand: PropTypes.func.isRequired,
+  onChangeMeasure: PropTypes.func.isRequired,
+  onChangeOrgUnit: PropTypes.func.isRequired,
+  selectedFilter: filterShape,
+  title: PropTypes.string,
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:109]:
```suggestion
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+};
+
+EntitySearchComponent.defaultProps = {
+  mobileListItems: [],
+  measureFilters: [],
+  measureFilterIsExpanded: false,
+  selectedFilter: null,
+  isLoading: false,
+  isMeasureLoading: false,
+  title: '',
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:119]:
```suggestion
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+  return (
+    <div>
+      <div>
+        <ExpandableList
+          items={mobileListItems.map(item => (
+            <SelectListItem onSelect={onChangeOrgUnit} {...item} />
+          ))}
+          expandedByDefault
+          title={title}
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:78]:
```suggestion
          title="Browse Entities"
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+  justify-content: center;
+  z-index: 2; // Above header.
+`;
+
+const NoResultsText = styled.div`
+  background-color: ${WHITE};
+  padding: ${MOBILE_MARGIN_SIZE}px;
+`;
+
+const EntitySearchComponent = ({
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:47]:
```suggestion
const EntityBrowseComponent = ({
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+        {mobileListItems.length === 0 && !isLoading && (
+          <NoResultsText>No clinics found in this region.</NoResultsText>
+        )}
+      </div>
+      {showBackButton && <BackButton orgUnit={orgUnit} />}
+    </div>
+  );
+};
+
+EntitySearchComponent.propTypes = {
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:98]:
```suggestion
EntityBrowseComponent.propTypes = {
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+  mobileListItems: PropTypes.array,
+  orgUnit: PropTypes.object.isRequired,
+  onToggleMeasureExpand: PropTypes.func.isRequired,
+  onChangeMeasure: PropTypes.func.isRequired,
+  onChangeOrgUnit: PropTypes.func.isRequired,
+  selectedFilter: filterShape,
+  title: PropTypes.string,
+};
+
+EntitySearchComponent.defaultProps = {
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:112]:
```suggestion
EntityBrowseComponent.defaultProps = {
```

File: packages/web-frontend/src/containers/mobile/EntityNav.js [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+    isMeasureLoading,
+    title,
+  };
+};
+
+export const EntityNav = connect(
+  mapStateToProps,
+  mapDispatchToProps,
+  mergeProps,
+)(EntitySearchComponent);
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/EntityNav.js:232]:
```suggestion
)(EntityBrowseComponent);
```

**Review by edmofro — CHANGES_REQUESTED** [review]
Have made some suggestions, but overall I also think the UX here also needs some work, because it would be unclear to the user how to select the entity and see their dashboards vs. just drill down to find the entity they're looking for. 

Would be great if we could separate out those two actions. 

It would also be really nice if we could make the existing SearchBar on Tupaia.org responsive and use that here, though I realise that sounds like potentially a lot more work for a quick win

**Comment by tcaiger** [comment]
Closing as needs design work

**Pull request closed without merging**